tests/test_simple_nlp.py

```diff
@@ -125,3 +127,92 @@ def test_simple_nlp_mlm(tiny_bert_tokenizer, tiny_bert_model):
     num_predict_batches_expected = ((size - 1) // batch_size) + 1
     assert len(predictions) == num_predict_batches_expected
     assert predictions[0].shape == (batch_size, sequence_length, vocab_size)
+
+@device('gpu')
+def test_simple_nlp_mlm_token_batch(tiny_bert_tokenizer, device):
+    transformers = pytest.importorskip('transformers')
+
+    vocab_size = tiny_bert_tokenizer.vocab_size
+    sequence_length = 32
+    size = 96
+    batch_size = 8
+    device = get_device(device)
+
+    train_dataset = RandomTextLMDataset(
+        size=size,
+        vocab_size=vocab_size,
+        sequence_length=sequence_length,
+        use_keys=True,
+        pad_token_id=tiny_bert_tokenizer.pad_token_id,
+    )
+    for i in range(size):  # Proactively load dataset
+        train_dataset[i]
```

Comment:
why is this necessary? Can we remove?

Reply:
This is necessary because the dataset only materializes its random tensors the first time they are invoked. So the first call to the trainer will initialize the dataset and result in a differently advanced random seed for e.g. random MLM masking.